src/api/app/controllers/webui/users_controller.rb

```diff
@@ -136,6 +144,28 @@ def change_password
 
   private
 
+  def adjust_filters
+    filters = params.slice(:search_text, :involved_projects, :involved_packages,
+                           :role_owner, :role_maintainer, :role_bugowner, :role_reviewer, :role_downloader, :role_reader)
+
+    filters[:search_text] = filters[:search_text]&.strip
+    @filters = filters.dup
+
+    if !filters.key?(:involved_projects) && !filters.key?(:involved_packages)
+      [:involved_projects, :involved_packages].each do |filter|
+        filters[filter] = 1
+      end
+    end
+    if !filters.key?(:role_owner) && !filters.key?(:role_maintainer) && !filters.key?(:role_bugowner) &&
+       !filters.key?(:role_reviewer) && !filters.key?(:role_downloader) && !filters.key?(:role_reader)
```

Comment:
You can also replace this condition with:
```ruby
roles = %i(role_owner role_maintainer role_bugowner role_reviewer role_downloader role_reader)
if (filters.keys & roles).empty?
```

When there is no intersection between the `filters.keys` array, that means the filters.keys array does not include any of the roles. Then we set the filters.

WDYT?

Reply:
That was nice. I applied your suggestion and moved the duplicated code into `set_all_filters_if_none_is_set`. Please have a look again.